File: java/src/main/java/org/licenselynx/LicenseLynx.java
/**
 * SPDX-FileCopyrightText: Copyright 2025 Siemens AG
 * SPDX-License-Identifier: Apache-2.0
 */
package org.licenselynx;

import javax.annotation.CheckForNull;

import javax.annotation.Nonnull;


/**
 * LicenseLynx class to map license names to their corresponding data from a JSON file.
 */
public final class LicenseLynx
{

    // Private constructor to prevent instantiation
    private LicenseLynx()
    {
    }



    /**
     * Maps the given license name to its corresponding LicenseObject.
     *
     * @param pLicenseName the name of the license to map
     * @return the license data as a LicenseObject, or null if not found
     */
    @CheckForNull
    public static LicenseObject map(@Nonnull final String pLicenseName)
    {
        LicenseMapSingleton licenseMapSingleton = LicenseMapSingleton.getInstance();
        LicenseMap licenseMap = licenseMapSingleton.getLicenseMap();

        return licenseMap.getCanonicalLicenseMap().get(pLicenseName);
    }



    /**
     * Maps the given license name to its corresponding LicenseObject.
     * It also searches through the risky license mappings, when the boolean value is set.
     *
     * @param pLicenseName the name of the license to map
     * @param pRisky boolean flag to enable risky mappings
     * @return the license data as a LicenseObject, or null if not found
     */
    @CheckForNull
    public static LicenseObject map(@Nonnull final String pLicenseName, final boolean pRisky)
    {
        LicenseMapSingleton licenseMapSingleton = LicenseMapSingleton.getInstance();
        LicenseMap licenseMap = licenseMapSingleton.getLicenseMap();

        LicenseObject licenseObject = licenseMap.getCanonicalLicenseMap().get(pLicenseName);

        if (licenseObject == null && pRisky) {
            licenseObject = licenseMap.getRiskyLicenseMap().get(pLicenseName);
        }

        return licenseObject;
    }
}

File: python/licenselynx/licenselynx.py
#
# SPDX-FileCopyrightText: Copyright 2025 Siemens AG
# SPDX-License-Identifier: Apache-2.0
#
import sys
from typing import Optional
from licenselynx.license_object import LicenseObject
from licenselynx.license_map_singleton import LicenseMapSingleton


class LicenseLynx:

    @staticmethod
    def map(license_name: str) -> Optional[LicenseObject]:
        """
        Maps license name to the canonical license identifier
        :param license_name: string of a license name
        :return: LicenseObject with the canonical license identifier and source, None if no license is found,
        or throws an exception if a runtime error occurs
        """
        try:

            instance = LicenseMapSingleton()

            entry = instance.merged_data.get(license_name, None)

            if not entry:
                return None

            return LicenseObject(canonical=entry.get("canonical"), src=entry.get("src"))
        except Exception as e:
            raise e.with_traceback(sys.exc_info()[2])
